### DISCUSSION TITLE
How to search within commits /git grep / git log)

### DISCUSSION THREAD

**Discussion by AtomicNess123** [magit/magit#4309] [Support] [answered]
I am using Magit and was wondering how to revert to a specific commit that contained a specific expression. First I need to find the expression across all commits content. How to do this? In command line it would be something like git grep / git log. Thanks!

**Comment by kyleam** [comment, marked as answer]
From the `magit-log` transient, you can use `-G` (search changes) or `-S` (search occurrences) to limit the log to matching commits.  `man git-log` has more information about the difference between `-G` and `-S`.  (And if you need to search commit _messages_, `magit-log` has `--grep` under `-F`.)

**Reply by AtomicNess123** [reply]
Thanks. When you say `magit-log transient`, what do you mean exactly? I do `l l `in the `magit-status` buffer and then `L`. Is that what you mean?

**Reply by tarsius** [reply]
Use `C-x w any-command RET` to find out what key(s) are bound to `any-command`. You will find that `l` is bound to `magit-log`.

Press `l` to invoke the `magit-log` command. It is a "transient command" so a buffer appears. That buffer lists actions, for example you can now press `l` to invoke the "show log for current" command. Instead of doing that directly you can first toggle switches and set options. They are shown above. If you type `-` and then `G` (aka `-G`), then you can set the value of the `-G` git argument.

This is documented. Start reading here: https://magit.vc/manual/magit/Transient-Commands.html.

**Reply by AtomicNess123** [reply]
Thanks! That was neatly explained. As for the C-c w, this keybinding is not bound to anything in my Emacs Prelude. What is the function name?

**Reply by tarsius** [reply]
I meant `C-h w`.

**Reply by AtomicNess123** [reply]
Thanks, that works!
